include/util/dynamic_graph.hpp

```diff
@@ -411,14 +413,19 @@ template <typename EdgeDataT> class DynamicGraph
         util::inplacePermutation(node_array.begin(), node_array.end(), old_to_new_node);
 
         // Build up edge permutation
-        auto new_edge_index = 0;
+        EdgeID new_edge_index = 0;
         std::vector<EdgeID> old_to_new_edge(edge_list.size(), SPECIAL_EDGEID);
         for (auto node : util::irange<NodeID>(0, number_of_nodes))
         {
             auto new_first_edge = new_edge_index;
             // move all filled edges
             for (auto edge : GetAdjacentEdgeRange(node))
             {
+                if (new_edge_index == std::numeric_limits<EdgeID>::max())
```

Comment:
Can this check be done before we start the loop and fail earlier?  We know the size of the edge_list, right?  This renumbering shouldn't create more than there was previously....

Reply:
yes, the check only prevents overflow of `number_of_valid_edges`, but you are right `old_to_new_edge` may be corrupted for dummy edges if `new_edge_index` overflows below